modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2Connection.java

```diff
@@ -59,6 +65,24 @@ public class H2Connection implements AutoCloseable {
         this.log = log;
 
         initStatementCache();
+
+        // Work around the H2 bug (NPE in Session#removeLocalTempTable).
+        // Make sure session always contains not-null list of temp tables.
+        try {
+            Session sess = (Session)(conn.unwrap(JdbcConnection.class)).getSession();
+
+            RangeTable dummyTbl = new RangeTable(
+                sess.getDatabase().getSchema(SCHEMA_MAIN),
+                ValueExpression.getNull(),
+                ValueExpression.getNull(),
+                true);
+
+            sess.addLocalTempTable(dummyTbl);
```

Comment:
Let's just create the collection using reflection. It will more clear, than creation/drop random table with implicit side effects.